core/templates/dev/head/services/AssetsBackendApiServiceSpec.js

```diff
@@ -59,32 +58,151 @@ describe('Assets Backend API Service', function() {
+      .image.length).toBe(1);
     $httpBackend.flush();
-
-    expect(successHandler).not.toHaveBeenCalled();
-    expect(failHandler).toHaveBeenCalled();
+    expect((AssetsBackendApiService.getAssetsFilesCurrentlyBeingRequested())
+      .image.length).toBe(0);
+    expect(AssetsBackendApiService.isCached('myfile.png')).toBe(true);
+    expect(successHandler).toHaveBeenCalled();
+    expect(failHandler).not.toHaveBeenCalled();
+    $httpBackend.verifyNoOutstandingExpectation();
   });
+
+  it('Should call the provided failure handler on HTTP failure for an audio',
+    function() {
+      var successHandler = jasmine.createSpy('success');
+      var failHandler = jasmine.createSpy('fail');
+
+      var requestUrl = UrlInterpolationService.interpolateUrl(
+        '/audiohandler/<exploration_id>/audio/<filename>', {
+          exploration_id: '0',
+          filename: 'myfile.mp3'
+        });
+
+      $httpBackend.expect('GET', requestUrl).respond(500, 'MutagenError');
+      AssetsBackendApiService.loadAudio('0', 'myfile.mp3').then(
+        successHandler, failHandler);
+      $httpBackend.flush();
+
+      expect(successHandler).not.toHaveBeenCalled();
+      expect(failHandler).toHaveBeenCalled();
+      $httpBackend.verifyNoOutstandingExpectation();
+    });
+
+  it('Should call the provided failure handler on HTTP failure for an image',
+    function() {
+      var successHandler = jasmine.createSpy('success');
+      var failHandler = jasmine.createSpy('fail');
+
+      var requestUrl = UrlInterpolationService.interpolateUrl(
+        '/imagehandler/<exploration_id>/image/<filename>', {
+          exploration_id: '0',
+          filename: 'myfile.png'
+        });
+
+      $httpBackend.expect('GET', requestUrl).respond(500, 'Error');
+      AssetsBackendApiService.loadImage('0', 'myfile.png').then(
+        successHandler, failHandler);
+      $httpBackend.flush();
+
+      expect(successHandler).not.toHaveBeenCalled();
+      expect(failHandler).toHaveBeenCalled();
+      $httpBackend.verifyNoOutstandingExpectation();
+    });
+
+  it('Should successfully abort the download of all the audio files',
+    function() {
+      var successHandler = jasmine.createSpy('success');
+      var failHandler = jasmine.createSpy('fail');
+
+      var requestUrl = UrlInterpolationService.interpolateUrl(
+        '/audiohandler/<exploration_id>/audio/<filename>', {
+          exploration_id: '0',
+          filename: 'myfile.mp3'
+        });
+
+      $httpBackend.expect('GET', requestUrl).respond(201, 'audio data');
+
+      AssetsBackendApiService.loadAudio('0', 'myfile.mp3').then(
+        successHandler, failHandler);
+
+      expect(AssetsBackendApiService.getAssetsFilesCurrentlyBeingRequested()
+        .audio.length).toBe(1);
+
+      AssetsBackendApiService.abortAllCurrentAudioDownloads();
+      try {
+        $httpBackend.flush();
+      } catch (exception) {
+        // the $httpBackend.flush throws the exception -- "There are no pending
```

Comment:
Dont do try/catch in tests.

If there are no pending requests to flush, why not use `verifyNoOutstandingRequest();` instead? See $httpBackend docs: https://docs.angularjs.org/api/ngMock/service/$httpBackend

Reply:
Okay! Done